### PR TITLE
external: fqdn should be persisted

### PR DESCRIPTION
**Pull request by parth-gr**
donot change rgw fqdn to ip if provided,
As now the bucket class supports the
entry of fqdn

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Review by travisn — COMMENTED** [review]
LGTM, were you able to validate this manually that fqdn can be used?

**Review by thotz — APPROVED** [review]
LGTM.

**Comment by parth-gr** [comment]
> LGTM, were you able to validate this manually that fqdn can be used?

Yaa we tested it.

**Comment by parth-gr** [comment]
Reconfirm from @BlaineEXE 

```
externalRgwEndpoints:
  - ip: Hostname 
   ```
  would work,
  
  Or should we change it like,
  ```
externalRgwEndpoints:
  - ip: ip
  - Hostname: hostname 
   ```

**Comment by parth-gr** [comment]
> There is legacy behavior that the `ip` field was used as a generic `address` based on the old k8s API that was used. `hostname` used to not be present, or it was not very useful, and I made changes to allow `hostname` to be a preferred field.
> 
> For the python scripts here, I think it's probably most clear for users to change the script to fill out the endpoints like:
> 
> ```
> externalRgwEndpoints:
>   - hostname: <hostname>
> ```
> 
> Even if the hostname is an IP, I think most users are familiar with putting IP into the hostname field but usually not the other way around.
> 
> I saw some users get confused by this recently leading to your work. I think it would be useful to add some comments to the API godocs to make sure no developers break the legacy behavior.
> 
> ```go
> // EndpointAddress is a tuple that describes a single IP address or hostname. This is a modified subset of
> // Kubernetes's v1.EndpointAddress.
> // +structType=atomic
> type EndpointAddress struct {
> 	// The IP of this endpoint. As a legacy behavior, this supports being given a DNS-adressable hostname as well.
> 	// +optional
> 	IP string `json:"ip" protobuf:"bytes,1,opt,name=ip"`
> 
> 	// The DNS-addressable Hostname of this endpoint. This field will be preferred over IP if both are given.
> 	// +optional
> 	Hostname string `json:"hostname,omitempty" protobuf:"bytes,3,opt,name=hostname"`
> }
> ```

THanks

File: deploy/examples/object-external.yaml [hunk 1]
```diff
@@ -16,3 +16,4 @@ spec:
     port: 80
     externalRgwEndpoints:
       - ip: 192.168.39.182
+        # hostname: example.com
```
Comment (thotz) [inline deploy/examples/object-external.yaml:19]: Add the Documentation as well https://rook.io/docs/rook/v1.11/Storage-Configuration/Object-Storage-RGW/object-storage/#connect-to-an-external-object-store

**Review by thotz — CHANGES_REQUESTED** [review]

File: deploy/examples/object-external.yaml (hunk 1 above)
Comment (parth-gr) [reply deploy/examples/object-external.yaml:19]: Added thanks

**Comment by parth-gr** [comment]
@BlaineEXE any suggestion,

I did run `make crds` and `make codegen` but the Crds test are still failing..

**Review by subhamkrai — COMMENTED** [review]
try to rebase your pr and then run the `make crds` and `make codegen` and see if that helps

**Review by thotz — APPROVED** [review]

**Review by travisn — CHANGES_REQUESTED** [review]
@parth-gr How about squashing to a single commit? And per discussion, open a separate issue to investigate the doc generation.

**Comment by Javlopez** [comment]
Hi @parth-gr

the command `make generate-docs-crds` only works when there are changes in `types.go`  using git diff

`git diff --name-only --diff-filter=M | grep -Ec '/types\.go$' ` so if there's no changes in git diff the command will skip the crds-docs generation,  we have another command

`make validate-and-gen-docs-crds` which basically is forcing the generation

```bash
$ make validate-and-gen-docs-crds
build/crds/generate-crd-docs.sh --force
differences found in types.go, rebuilding specification.md
I0531 09:59:29.112480   73922 main.go:129] parsing go packages in directory github.com/rook/rook/pkg/apis/ceph.rook.io
I0531 09:59:44.041123   73922 main.go:231] using package=github.com/rook/rook/pkg/apis/ceph.rook.io/v1
W0531 09:59:44.056995   73922 main.go:442] not found external link source for type time.Duration
W0531 09:59:44.060464   73922 main.go:442] not found external link source for type time.Duration
W0531 09:59:44.062225   73922 main.go:442] not found external link source for type time.Duration
W0531 09:59:44.062263   73922 main.go:442] not found external link source for type time.Duration
W0531 09:59:44.064344   73922 main.go:442] not found external link source for type k8s.io/apimachinery/pkg/api/resource.Quantity
W0531 09:59:44.066305   73922 main.go:442] not found external link source for type k8s.io/apimachinery/pkg/api/resource.Quantity
I0531 09:59:44.076182   73922 main.go:167] written to /home/javier/src/rook/Documentation/CRDs/specification.md
```

You can run that command instead to generate.

My suggestion is to add some message like this

```bash
$make generate-docs-crds
No changes found in types.go, you can force the crds document generation by running 
make validate-and-gen-docs-crds
```

Another option worth considering is to consistently run the CRDs generation process everytime I mean without git.
 
Although this approach may introduce a slight delay of a few seconds, I believe the benefits outweigh this cost.

Any thoughts?

we can follow the discussion here https://github.com/rook/rook/issues/12310 in order to find the best solution?

**Pull request merged**